lib/grandpa/network.go

```diff
@@ -222,8 +226,14 @@ func decodeMessage(cm *network.ConsensusMessage) (m GrandpaMessage, err error) {
 		m = &val
 	case CommitMessage:
 		m = &val
-	case NeighbourMessage:
-		m = &val
+
+	case VersionedNeighbourPacket:
+		switch NeighbourMessage := val.Value().(type) {
+		case NeighbourPacketV1:
+			m = &NeighbourMessage
+		default:
+			return nil, ErrInvalidMessageType
```

Comment:
A bit of a new thing, we should wrap sentinel errors even without context, so it forces callers to use `errors.Is` instead of `==` which would fail later if some context is added.

```suggestion
			return nil, fmt.Errorf("%w", ErrInvalidMessageType)
```